### ISSUE TITLE
Using netopper2 for configuraing nerwork interfaces

### ISSUE THREAD

**Issue by alireza11048** [CESNET/netopeer2#1239]
Hi,
I've installed and configured the netopeer2, and I've managed to use it to transmit/retrieve some configuration through CLI to netopeer server.
I can see the ietf-interfaces model is installed by default in the sysrepo. my question is, can I use this setup to configure device interfaces through CLI, for example setting the IP address for an interface or setting the gateway? or any extra app should be installed? is there any tutorial about this?

thanks in advance

**Comment by jktjkt** [comment]
sysrepo won't install `ietf-interfaces`; there's a test suite which uses that YANG model, but it should run in a sandbox with a separate storage location. What version are you using?

Also, sysrepo itself is "just a YANG database", it won't really control the system on its own. For that, you need applications which *use* sysrepo and configure your system in response to *changes* of the sysrepo DB content -- perhaps as edited via NETCONF through Netopeer2. There are several open source projects which build on top of the `ietf-interfaces` YANG model:
- [`CESNET/velia`](https://github.com/CESNET/velia)
- [`telekom/sysrepo-plugin-ietf-interfaces`](https://github.com/telekom/sysrepo-plugin-interfaces)

You can find more [on the homepage](https://www.sysrepo.org/plugins). Also, I recommend reading the docs which explain what exactly is sysrepo.

**Comment by jktjkt** [comment]
> Hi, is the problem solved? I have the similar question,could you tell me how to configure?

Have you read my comment above?

**Comment by 1061136002** [comment]
> I recommend reading the docs which explain what exactly is sysrepo

Yes, I still don't know how to fix it. Can you explain more specifically, thanks.

**Comment by jktjkt** [comment]
As explained in #1410, it looks that there's nothing to fix here. This project (Netopeer2 and sysrepo) is all about a "NETCONF middleware". That's something that developers use to solve whatever problem needs solving -- as long as a part of that problem is "using NETCONF" or "using YANG".

It seems to me that the problem you're trying to solve is "network configuration through NETCONF". That means that you're probably looking for a tool which *uses* this middleware for configuring your network. Now, I've [linked to two projects](https://github.com/CESNET/netopeer2/issues/1239#issuecomment-1186552702) which *use* this middleware for just that. You can either pick one of them, or you can write some code which *uses* this middleware and fits your purposes better.

To offer you an analogy, suppose you need an e-shop. You won't ask the vendor of your SQL DB server to "implement an e-shop". You look for some SW which *uses* the SQL server and which happens to implement an e-shop, or you write your own SW which *uses* that SQL DB server. But it's not a job of that SQL server to implement an e-shop, and it's not a job of Netopeer2 or sysrepo to configure network interfaces of a Linux server.

**Comment by 1061136002** [comment]
> As explained in #1410, it looks that there's nothing to fix here. This project (Netopeer2 and sysrepo) is all about a "NETCONF middleware". That's something that developers use to solve whatever problem needs solving -- as long as a part of that problem is "using NETCONF" or "using YANG".
> 
> It seems to me that the problem you're trying to solve is "network configuration through NETCONF". That means that you're probably looking for a tool which _uses_ this middleware for configuring your network. Now, I've [linked to two projects](https://github.com/CESNET/netopeer2/issues/1239#issuecomment-1186552702) which _use_ this middleware for just that. You can either pick one of them, or you can write some code which _uses_ this middleware and fits your purposes better.
> 
> To offer you an analogy, suppose you need an e-shop. You won't ask the vendor of your SQL DB server to "implement an e-shop". You look for some SW which _uses_ the SQL server and which happens to implement an e-shop, or you write your own SW which _uses_ that SQL DB server. But it's not a job of that SQL server to implement an e-shop, and it's not a job of Netopeer2 or sysrepo to configure network interfaces of a Linux server.

Thank you, I understand what you are trying to say.
